tests/testthat/test-actual-distributions.R

```diff
@@ -0,0 +1,45 @@
+library(hedgehog)
+freeze_eval <- names(.GlobalEnv)
+
+test_that("beta distributed data are generated correctly", {
+  skip_on_cran()
+  
+  p1 <- runif(1, .1, .9)
+  v1 <- p1*(1-p1)/2
+  
+  p2 <- runif(1, .1, .9)
+  v2 <- p2*(1-p2)/2
+  
+  logitp2 <- log(p2 / (1-p2))
+  def <- defData(varname = "b1", formula = "..p1", variance = 1, dist = "beta")
+  def <- defData(def, varname = "b2", formula = "..logitp2", variance = 1, 
+                 dist = "beta", link="logit")
+  
+  dd <- genData(n = 1000, dtDefs = def)
+  diff <- dd[, .(abs(mean(b1) - p1), abs(mean(b2) - p2)) ]
+  expect_true(all(diff < 0.05))
+  
+  diffv <- dd[, .(abs(var(b1) - v1), abs(var(b2) - v2)) ]
+  expect_true(all(diffv < 0.02))
```

Comment:
I have not looked through this thoroughly, would need some time but trust you have this aspect covered :+1:

Reply:
Not sure I would trust me, but I did vet this pretty thoroughly.